lib/logstash/filters/xml.rb

```diff
@@ -203,6 +203,14 @@ def filter(event)
 
     filter_matched(event) if matched
     @logger.debug? && @logger.debug("Event after xml filter", :event => event)
+  rescue Exception => e
```

Comment:
any specific reason to rescue from `Exception` instead of the default `rescue => e` that is equivalent to `rescue StandardError => e` ?

Reply:
(to add: in general we want to avoid rescuing from `Exception` because that also includes exceptions such as `SyntaxError` and `NoMemoryError` so by rescuing from `Exception` we are actually broadening the rescue scope from the default `rescue => e`)

Reply:
The specific exception that sent me down this path was a `org.w3c.dom.DOMException`, which I _believe_ jruby wraps in a proxy subclass of `(org.jruby.Ruby)Exception`. I'm a little fuzzy on the java/ruby bridge here, and figured this would be the safest way of catching anything that was catchable on the Ruby side.

Reply:
Ah, Java exceptions. Ok, +1 on moving this forward like this to make sure it works and I suggest we open a LS issue to revisit the rescuing strategy for making sure we rescue Java exceptions without having to broaden the rescue scope unnecessarily. The last work on this front is quite old and we should revisit IMO (ref https://github.com/elastic/logstash/issues/4064 )